src/tools/cephfs_mirror/Mirror.cc

```diff
@@ -509,16 +510,20 @@ void Mirror::update_fs_mirrors() {
       auto failed = mirror_action.fs_mirror && mirror_action.fs_mirror->is_failed();
       auto blocklisted = mirror_action.fs_mirror && mirror_action.fs_mirror->is_blocklisted();
 
-      if (check_failure && !mirror_action.action_in_progress && failed) {
+      if (check_failure && !mirror_action.action_in_progress &&
+	  !_is_restarting(filesystem) && failed) {
         // about to restart failed mirror instance -- nothing
         // should interfere
         dout(5) << ": filesystem=" << filesystem << " failed mirroring -- restarting" << dendl;
+	_set_restarting(filesystem);
         auto peers = mirror_action.fs_mirror->get_peers();
         auto ctx =  new C_RestartMirroring(this, filesystem, mirror_action.pool_id, peers);
         ctx->complete(0);
-      } else if (check_blocklist && !mirror_action.action_in_progress && blocklisted) {
+      } else if (check_blocklist && !mirror_action.action_in_progress &&
+		 !mirror_action.restarting && blocklisted) {
```

Comment:
MirrorAction struct doens't have a `restarting` member, i think you meant `!_is_restarting(filesystem)` here

Reply:
stale code - fixing...

Reply:
Fixed -- the helper is used in the `if {}` block.

Reply:
why not just use `!mirror_action.restarting` everywhere

Reply:
that's fine too, but since we have set/unset helpers, it made sense to add the `is_` check.

Reply:
we already have mirror_action from the for loop the `if {}` block is in, using a helper would go to the m_mirror_actions map to find the same mirror_action again. i feel this can be avoided.

Reply:
> we already have mirror_action from the for loop the `if {}` block is in, using a helper would go to the m_mirror_actions map to find the same mirror_action again. i feel this can be avoided.

That's a minor thing as compared with the readability and more importantly in C_RestartMirroring the flag needs to be unset

```
   void handle_enable_mirroring(int r) {
     mirror->handle_enable_mirroring(filesystem, peers, r);
+    mirror->_unset_restarting(filesystem);
     delete this;
   }
```